src/ingestion/models/ingestionManager.ts

```diff
@@ -173,6 +176,133 @@ export class IngestionManager {
+      case true:
+        return this.handleRetryWithoutErrors(jobId, validationTask.id, logCtx);
+      default: {
+        const msg = 'Cannot retry job because validation task status is unclear';
+        this.logger.error({ msg, logContext: logCtx, jobId, taskId: validationTask.id });
+        const error = new BadRequestError(msg);
+        trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+        throw error;
+      }
+    }
+  }
+
+  @withSpanV4
+  private validateRetryJobStatus(jobId: string, status: OperationStatus, logCtx: LogContext): void {
+    const validStatuses = [OperationStatus.FAILED, OperationStatus.COMPLETED];
+    if (!validStatuses.includes(status)) {
+      const message = `Cannot retry job with id: ${jobId} because its status is: ${status}. Expected status: ${validStatuses.join(' or ')}`;
+      this.logger.error({ msg: message, logContext: logCtx, jobId, currentStatus: status, validStatuses });
+      const error = new BadRequestError(message);
+      trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+      throw error;
+    }
+  }
+
+  @withSpanAsyncV4
+  private async getValidationTask(jobId: string, logCtx: LogContext): Promise<ITaskResponse<ValidationTaskParameters>> {
+    const tasks = await this.jobManagerWrapper.getTasksForJob<ValidationTaskParameters>(jobId);
+
+    const validationTask = tasks.find((task) => task.type === this.validationTaskType);
+
+    if (!validationTask) {
+      const message = `Cannot retry job with id: ${jobId} because no validation task was found`;
+      this.logger.error({ msg: message, logContext: logCtx, jobId, taskTypes: tasks.map((t) => t.type) });
+      const error = new NotFoundError(message);
+      trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+      throw error;
+    }
+
+    return validationTask;
+  }
+
+  @withSpanAsyncV4
+  private async handleRetryWithoutErrors(jobId: string, taskId: string, logCtx: LogContext): Promise<ResponseId> {
+    this.logger.info({ msg: 'validation completed without errors, resetting job', logContext: logCtx, jobId, taskId });
+    await this.jobManagerWrapper.resetJob(jobId);
+    trace.getActiveSpan()?.setStatus({ code: SpanStatusCode.OK }).addEvent('ingestionManager.retryLayer.success', { retryType: 'reset', jobId });
+    this.logger.info({ msg: 'job reset successfully', logContext: logCtx, jobId, taskId });
+    return { jobId, taskId };
+  }
+
+  @withSpanAsyncV4
+  private async handleRetryWithErrors(
+    jobId: string,
+    retryJob: { parameters: z.infer<typeof ingestionBaseJobParamsSchema> },
+    validationTask: ITaskResponse<ValidationTaskParameters>,
+    logCtx: LogContext
+  ): Promise<ResponseId> {
+    this.logger.info({ msg: 'validation has errors, checking for shapefile changes', logContext: logCtx, jobId, taskId: validationTask.id });
+
+    const metadataShapefilePath = retryJob.parameters.inputFiles.metadataShapefilePath;
```

Comment:
please make a Zod parse validation for input files schema here to make sure if there are no missing necessary  job parameters